internal/services/repository/data_source_repository_npm_group.go

```diff
@@ -0,0 +1,30 @@
+package repository
+
+import (
+	"github.com/datadrivers/terraform-provider-nexus/internal/schema/common"
+	"github.com/datadrivers/terraform-provider-nexus/internal/schema/repository"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceRepositoryNpmGroup() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this data source to get an existing npm group repository.",
+
+		Read: dataSourceRepositoryNpmGroupRead,
+		Schema: map[string]*schema.Schema{
+			// Common schemas
+			"id":     common.DataSourceID,
+			"name":   repository.DataSourceName,
+			"online": repository.DataSourceOnline,
+			// Group schemas
+			"group":   repository.DataSourceGroup,
```

Comment:
please use repository.DataSourceGroupDeploy

Reply:
if you don't have a nexus pro license, i can implement this after merging your pull request.

Reply:
extract into new ticket